src/reducers/players.py

```diff
@@ -15,8 +18,27 @@ def players(state=None, action=None):
     if action is None:
         return state
 
-    ty = action.get('type')
-    if ty == 'CREATE_PLAYER':
-        return state + [create_player(action.get('player_id'))]
+    action_type = action.get('type')
+    if action_type == 'PLAYER_CHANGE_MONEY':
+        next_state = deepcopy(state)
+        for p in next_state:
+            if p == action.get('player_id'):
+                p['money'] += action.get('money')
+        return next_state
+    if action_type == 'PLAYER_ADD_POWER_PLANT':
+        next_state = deepcopy(state)
+        for p in next_state:
+            if p == action.get('player_id'):
+                if len(p['power_plants']) < 4:
+                    p['power_plants'].append('power_plant')
```

Comment:
This just appends the string `power_plant` onto the power_plants list.  It needs to be something like this:
```python
p['power_plants'].append(action.get('power_plant'))
```